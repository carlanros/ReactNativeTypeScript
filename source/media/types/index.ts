// media/types/index.ts

export enum MediaConstants {
  /**
   * Login failed or was cancelled
   */
  MEDIA_FAILED = 'MEDIA_FAILED',
  /**
   * Login state: authentication succeeded
   *
   * @static
   * @memberof MediaConstants
   */
  MEDIA_SUCCESS = 'MEDIA_SUCCESS',

  /**
   * Login state: authentication in progress
   *
   * @static
   * @memberof MediaConstants
   */
  MEDIA_BUSY = 'Kollar vem du Ã¤r',
}

export interface IMediaState {
  cancelled: boolean;
  height: number;
  width: number;
  uri: string;

}
export interface IMediaAction extends IMediaState {
  type: MediaConstants;
}
