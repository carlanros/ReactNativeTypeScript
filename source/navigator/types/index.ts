export interface INavigationState {
  /**
   * Current scene object in navigation
   *
   * @type {object}
   * @memberof INavigationReducer
   */
  scene: object;
}

export interface INavigationAction extends INavigationState {
  type: string;
}
