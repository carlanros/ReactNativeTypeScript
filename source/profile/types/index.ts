// profile/types/index
import * as types from '../../Types';

export interface IProfileProps extends types.IProps {
  Logout?: () => any;
}
