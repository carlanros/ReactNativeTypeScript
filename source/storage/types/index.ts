// storage/types/index

/**
 * App storage Redux action types
 * @type {StorageConstants}
 * @public
 * @type {Object}
 */
export enum StorageConstants {
    STORAGE_GOTKEY = 'STORAGE = GOT KEY',
    STORAGE_REMOVEDKEY = 'STORAGE = REMOVED KEY',
    STORAGE_REMOVED_FAILED = 'STORAGE = REMOVED KEY FAILED',
    STORAGE_NOSUCHKEY = 'STORAGE = NO SUCH KEY',
    STORAGE_SAVE_FAILED = 'STORAGE = SAVE FAILED',
    STORAGE_SAVED = 'STORAGE = SAVED KEY',
    STORAGE_USER_KEY = 'STORAGE KEY = USER',
    STORAGE_BUSY = 'Friskar upp minnet',
  }

export interface IKey {
/**
 * The key in app storage
 *
 * @type {string}
 */
  key: string;
}

/**
 * Reducer properties passed down from StorageReducer
 * @typedef IStorageReducer
 * @type {Object}
 * @public
 * @export
 * @property {Object} value - the app storage value
 * @property {string} key - The app storage key
 */
export interface IStorageState extends IKey {

  /**
   * Value to save / load from app storage
   *
   * @type {object?}
   */
  value?: object;
}

export interface IStorageAction extends IStorageState {
  type: StorageConstants;
}