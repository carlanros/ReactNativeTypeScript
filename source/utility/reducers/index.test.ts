// utility/reducers/index.test

import AppReducer from './';
import * as types from '../types';

const senderFunc = 'AppReducer test';

describe('Utility Reducer', () => {
  const busyReason = 'Busy testing';

  test('should return initial state', () => {
    expect(AppReducer(null, { type: null })).toMatchSnapshot();
  });

  test(`should handle ${types.UtilityConstants.APP_LOAD_BUSY}`, () => {
    expect(
      AppReducer(null, {
        type: types.UtilityConstants.APP_LOAD_BUSY,
        reason: busyReason,
        sender: senderFunc,
      })
    ).toEqual({
      isBusy: true,
      busyReason: busyReason,
      sender: senderFunc,
    });
  });

  test(`should handle ${types.UtilityConstants.APP_LOAD_DONE}`, () => {
    expect(
      AppReducer(null, {
        type: types.UtilityConstants.APP_LOAD_DONE,
        sender: senderFunc,
      })
    ).toEqual({
      sender: senderFunc,
      isBusy: false,
      busyReason: null,
    });
  });

  test(`should handle ${types.UtilityConstants.APP_HAS_ERROR}`, () => {
    expect(
      AppReducer(null, {
        type: types.UtilityConstants.APP_HAS_ERROR,
        reason: 'error',
        exception: 'ex',
        sender: senderFunc,
      })
    ).toEqual({
      exception: 'ex',
      sender: senderFunc,
      hasError: true,
      errorMessage: 'error',
    });
  });

  test(`should handle ${types.UtilityConstants.APP_NO_ERROR}`, () => {
    expect(
      AppReducer(null, {
        type: types.UtilityConstants.APP_NO_ERROR,
        sender: senderFunc,
      })
    ).toEqual({
      errorMessage: null,
      exception: null,
      sender: senderFunc,
      hasError: false,
    });
  });
});
