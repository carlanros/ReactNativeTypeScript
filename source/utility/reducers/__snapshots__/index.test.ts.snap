// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utility Reducer should return initial state 1`] = `null`;
